pkg/stub/handler_test.go

```diff
@@ -451,6 +453,95 @@ func TestImageStreamEvent(t *testing.T) {
 	validate(true, err, "", cfg, conditions, statuses, t)
 }
 
+func TestImageStreamErrorRetry(t *testing.T) {
+	h, cfg, event := setup()
+	processCred(&h, cfg, t)
+	mimic(&h, x86OCPContentRootDir)
+	err := h.Handle(event)
+	statuses := []corev1.ConditionStatus{corev1.ConditionFalse, corev1.ConditionTrue, corev1.ConditionTrue, corev1.ConditionTrue, corev1.ConditionFalse, corev1.ConditionFalse, corev1.ConditionFalse}
+	validate(true, err, "", cfg, conditions, statuses, t)
+	// expedite the stream events coming in
+	cache.ClearUpsertsCache()
+
+	tagVersion := int64(1)
+	is := &imagev1.ImageStream{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+			Annotations: map[string]string{
+				v1.SamplesVersionAnnotation: h.version,
+			},
+		},
+		Spec: imagev1.ImageStreamSpec{
+			Tags: []imagev1.TagReference{
+				{
+					Name:       "foo",
+					Generation: &tagVersion,
+					From: &corev1.ObjectReference{
+						Kind: "DockerImage",
+						Name: "foofoo",
+					},
+				},
+			},
+		},
+		Status: imagev1.ImageStreamStatus{
+			Tags: []imagev1.NamedTagEventList{
+				{
+					Tag: "foo",
+					Conditions: []imagev1.TagEventCondition{
+						{
+							Generation: tagVersion,
+							Status: corev1.ConditionFalse,
+							Message: "failed import",
+						},
+					},
+				},
+			},
+		},
+	}
+
+	h.processImageStreamWatchEvent(is, false)
+
+	if !cfg.ConditionTrue(v1.ImportImageErrorsExist) {
+		t.Fatalf("Import Error Condition not true: %#v", cfg)
+	}
+
+	importError := cfg.Condition(v1.ImportImageErrorsExist)
+	initialImportErrorLastUpdateTime := cfg.Condition(v1.ImportImageErrorsExist).LastUpdateTime
+	h.processImageStreamWatchEvent(is, false)
+	// refetch to see if updated
+	importError = cfg.Condition(v1.ImportImageErrorsExist)
```

Comment:
```suggestion
	importError := cfg.Condition(v1.ImportImageErrorsExist)
```

(you never used the original copy above).